libgeopmd/src/LevelZeroDevicePool.cpp

```diff
@@ -266,12 +266,35 @@ namespace geopm
                                             dev_subdev_idx_pair.second);
     }
 
+    static double convert_active_timestamp(uint64_t value, uint64_t &last_value, uint64_t &rollover_count, uint64_t &original_value)
+    {
+        static const int num_bits = 32;
+        static const uint64_t overflow = (1ULL << num_bits);
+        static const double overflow_d = overflow;
+        static const uint64_t mask = overflow - 1;
+
+        value &= mask;
+        if (original_value > value) {
+            ++rollover_count;
+        }
+        if (value == original_value) {
+            value = last_value + 1;
```

Comment:
Add a comment here explaining that this is required because when there is no activity, the counter is not incrementing.
```suggestion
            // There are time and timestamp signals associated with all activity signals.
            // Time and timestamp do not change if there is no activity, however the sample()
            // API requires that the timstamp signal value increase on each call.
            value = last_value + 1;
```